homeassistant/components/qbittorrent/__init__.py

```diff
@@ -21,7 +21,7 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORMS = [Platform.SENSOR]
+PLATFORMS = [Platform.SENSOR, Platform.SWITCH]
 
 
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
```

Comment:
I would recommend you pass the result of `client.get_alternative_speed_status` in `setup_client` and feed that to the entity setup.

The way I did it was to put this in the coordinator.
```python
def set_alt_speed_enabled(self, is_enabled: bool) -> None:
    """Set the alternative speed mode."""
    if self.get_alt_speed_enabled() != is_enabled:
        self.toggle_alt_speed_enabled()

def toggle_alt_speed_enabled(self) -> None:
    """Toggle the alternative speed mode."""
    self.client.toggle_alternative_speed()

def get_alt_speed_enabled(self) -> bool:
    """Get the alternative speed mode."""
    return self._is_alternative_mode_enabled
```

And have these as my entity description
```python
@dataclass(frozen=True)
class QBittorrentSwitchEntityDescriptionMixin:
    """Mixin for required keys."""

    is_on_func: Callable[[QBittorrentDataCoordinator], bool | None]
    on_func: Callable[[QBittorrentDataCoordinator], None]
    off_func: Callable[[QBittorrentDataCoordinator], None]
    toggle_func: Callable[[QBittorrentDataCoordinator], None]

...

        is_on_func=lambda coordinator: coordinator.get_alt_speed_enabled(),
        on_func=lambda coordinator: coordinator.set_alt_speed_enabled(True),
        off_func=lambda coordinator: coordinator.set_alt_speed_enabled(False),
        toggle_func=lambda coordinator: coordinator.toggle_alt_speed_enabled(),
```


That way the initial value is correct.

Reply:
Updated with the change you suggested, can you take a look ?